File: app/models/file.server.model.java
'use strict';

/**
 * Module dependencies.
 */
var mongoose = require('mongoose'),
	Schema = mongoose.Schema;


var FileSchema = new Schema({
	fileName: {
		type: String,
		trim: true,
		required: 'Please fill keyword',
		default: ''
	},
	fileKeyword: {
		type: String, 
		trim: true,
		required: 'Enter some keywords',
		default: ''
	},
	fileAuthor: {
		type: String, 
		required: 'Enter some author',
		default: 'NA'
	},
	fileDate: {
		type: Date, 
	},
	fileExt: {
		type: String, 
	}
	fileNotes: {
		type: String,
	}

});

mongoose.model('File', FileSchema);

File: app/models/file.server.model.js
'use strict';

/**
 * Module dependencies.
 */
var mongoose = require('mongoose'),
	Schema = mongoose.Schema;

/**
 * File Schema
 */
var FileSchema = new Schema({
	title: {
		type: String,
		default: '',
		required: 'Please fill File name',
		trim: true
	},
	fileType: {
		type: String,
		default: '',
		required: 'File type needed'
	},
	notes: {
		type: String, 
		default: ''
	},
	author: {
		type: String,
		default: ''
	},
	keywords: {
		type: String, 
		default: ''
	},
	description: {
		type: String,
		default: ''
	},
	abstract: {
		type: String, 
		default: ''
	},
	created: {
		type: Date,
		default: Date.now
	},
	user: {
		type: Schema.ObjectId,
		ref: 'User'
	},
	publisherLocation: {
		type: String, 
		default: ''
	},
	publisher:{
		type: String, 
		default: ''
	},
	year: {
		type: String, 
		default: ''
	},
	ifEdit:{
		type: Date
	},
	fileBlob:{
		type: Buffer
	}
});

mongoose.model('File', FileSchema);